eZ/Publish/Core/Persistence/Legacy/Filter/CriterionQueryBuilder/Content/ObjectStateIdQueryBuilder.php

```diff
@@ -29,19 +29,23 @@ public function buildQueryConstraint(
         FilteringQueryBuilder $queryBuilder,
         FilteringCriterion $criterion
     ): ?string {
+        static $counter = 1;
+        $tableAlias = 'object_state_link_' . $counter;
+        ++$counter;
```

Comment:
I'm not a huge fan of static variables. There's no way to know for how long the object will live (maybe it's a batch process?) and if it performs multiple queries then the counter increases even though it doesn't need to.

I'd use something like: 
```suggestion
        $tableAlias = uniqid('osl_');
```
to ensure every alias is unique.

Reply:
The precision of uniqid() is microseconds, might not be good enough ? : https://www.php.net/manual/en/function.uniqid.php#120123

Precision is increased by using second argument to function ($more_entropy), but that creates IDs in format `12345678.1234567`, and the dot would then need to be removed as it is not accepted in table aliases in sql

Some systems requires the use of `$more_entropy`, not limited to Cygwin....

Still okay with your suggestion, @alongosz  ?

Reply:
`uniqid` increases the internal counter if called multiple times. It ensure no collision, unless between multiple PHP processes.

Reply:
Behavior described in the comment in PHP docs does not seem to happen. It might be something that is specific to some environments.

Current `uniqid` implementation ensures that at least a milisecond has passed since generating ID, which makes it almost impossible to hit the same sequence twice.

https://github.com/php/php-src/blob/b1f5ca59d188337e4545f2e8df3d0e6936299fe4/ext/standard/uniqid.c#L55-L62.

This code has 7 years, so you can safely assume it holds true for our supported versions :sweat_smile:

Reply:
Roger. Applied suggestion. Thank you